chips/nrf51/src/uart.rs

```diff
@@ -142,7 +159,25 @@ impl UART {
             }
         }
         if tx {
-            // Should never execute!
+            regs.event_txdrdy.set(0 as u32);
+
+            if self.len.get() == self.index.get() {
+                regs.task_stoptx.set(1 as u32);
+
+                // Signal client write done
+                match self.client {
```

Comment:
```rust
self.client.map(|client| {
   ...
});
```

Would accomplish this more cleanly than a `match` with an empty `None` condition (it's actually exactly the same thing)